### PR TITLE
Allow limiting the size of the dataloaders from the CLI

### PR DESCRIPTION
**Pull request by MJHamar**
use the argument 
`--max_len_samples <k>`

### CODE REVIEW

File: adapt_med_seg/data/dataset.py [hunk 1]
```diff
@@ -231,13 +232,20 @@ def _load_single_dataset(self, data_json: str, splits: List[str|Tuple[str,str]])
...
+        max_len_samples: int = None
     ) -> tuple[DataLoader, DataLoader]:
         if not self.train:
             raise ValueError("This method is only for training dataset")
 
         train_subset = self._tr_val_splits["training"]
         val_subset = self._tr_val_splits["validation"]
+        if (max_len_samples is not None
+            and max_len_samples < len(train_subset)
+            and max_len_samples < len(val_subset)):
```
Comment (MJHamar) [inline adapt_med_seg/data/dataset.py:245]: This part here might be too restricting, because if the val set is small then this condition may not be true in many cases. Depends on what are the use cases

Comment (MJHamar) [reply adapt_med_seg/data/dataset.py:245]: lmk if I should fix it

Comment (MJHamar) [reply adapt_med_seg/data/dataset.py:245]: then the get_test_dataloader calls should not pass this argument. Leaving it like this for now but keep in mind

**Comment by MJHamar** [comment]
I couldn't let go of the known issues so I added a separate argument for limiting the size of the test set 
`--max_num_test_samples <k>` so we don't get surprised
Also moved the train and val dataloader construction to separate if statements, so that if the validation set is smaller than the required maximum, the train set will still be limited
